src/py/flwr/server/driver/driver.py

```diff
@@ -69,44 +70,185 @@ def _get_grpc_driver_and_run_id(self) -> Tuple[GrpcDriver, int]:
             self.grpc_driver.connect()
             res = self.grpc_driver.create_run(CreateRunRequest())
             self.run_id = res.run_id
-
         return self.grpc_driver, self.run_id
 
-    def get_nodes(self) -> List[Node]:
+    def _check_message(self, message: Message) -> None:
+        # Check if the message is valid
+        if not (
+            message.metadata.run_id == self.run_id
+            and message.metadata.src_node_id == self.node.node_id
+            and message.metadata.message_id == ""
+            and message.metadata.reply_to_message == ""
+        ):
+            raise ValueError(f"Invalid message: {message}")
```

Comment:
There should be a test for this. I should verify if an exception is being raised by using `push_messages`. It would pass valid and invalid messages to it and test if an exception is raised.

Reply:
Thanks. I've added a test.